content/articles/creating-serverless-functions-faas-with-aws-lambda-and-golang/index.md

```diff
@@ -0,0 +1,292 @@
+
+### Setting up a Lambda Function on AWS
+
+<<<<<<< HEAD
+1. After Creating your AWS account, on the main dashboard, search for Lambda.![search-aws-lambda](search-aws-dashboard.png)When you click on Lamda, the AWS lambda will be visible.![aws-lambda-dashboard](aws-lambda-dashboard.png)
+2. Click on the Create Function button to set up a new AWS Lambda function.![aws-create-lambda-setting](aws-create-lambda-setting.png)The name of the function is user-profile. Click on Create function.
+3. On the function’s dashboard, we will upload the function we want to run in the cloud. ![aws-lambda-function-dashboard](aws-lambda-function-dashboard.png)In the next section, we'll look at how to make a function that runs on AWS Lambda.
+=======
+1. After Creating your AWS account, on the main dashboard, search for Lambda.![search-aws-lambda](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/search-aws-dashboard.png)When you click on Lamda, you will see the AWS lambda page.![aws-lambda-dashboard](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-dashboard.png)
+2. Click on the Create Function button to set up a new AWS Lambda function.![aws-create-lambda-setting](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-create-lambda-setting.png)The name of the function is user-profile. Click on Create function.
+3. On the function’s dashboard, we will upload the function we want to run in the cloud. ![aws-lambda-function-dashboard](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-function-dashboard.png)In the next section, we will see how to create a function that will run on AWS lambda.
+>>>>>>> 012ab3bce84482631c2308ae86a7bca9c5bbe029
+
+### Creating the User-Profile Function in Golang
+
+In this section, we will see how to make the function that AWS Lambda will run and how to convert it into a zip file before we upload it on AWS Lambda. We need to install the "aws-lambda-go" package with:
+
+* `go mod init`
+* `go get github.com/aws/aws-lambda-go/lambda`
+
+```go
+
+package main
+
+ 
+
+import (
+
+ "fmt"
+
+)
+
+ 
+
+type InfoEvent struct {
+
+ Firstname string `json:"firstname"`
+
+ Lastname string `json:"lastname"`
+
+ Age int `json:"age"`
+
+}
+
+ 
+
+type Response struct {
+
+ Profile string `json:"profile"`
+
+}
+
+ 
+
+func HandleInfoEvent(event InfoEvent) (Response, error) {
+
+ return Response{Profile: fmt.Sprintf("Their name is %s %s, they are %d ", event.Firstname, event.Lastname, event.Age)}, nil
+
+}
+
+ 
+
+func main() {
+
+ lambda.Start(HandleInfoEvent)
+
+}
+
+```
+
+You can also run `git clone https://github.com/Bamimore-Tomi/faas-golang.git` and navigate to the “server” folder to see this source code. In the program above, we created the `HandleInfoEvent` function. This is the function that runs in the cloud. Use the following steps to compile and zip the program.
```

Comment:
`You can also run `git clone https://github.com/Bamimore-Tomi/faas-golang.git``

You can add this either during the start of the tutorial, and walk the reader through the concepts simultaneously. Or explain all the concepts, and ask the reader to refer this for more info.

Reply:
Done, I have added it at the start of the tutorial. Thank you.